### PR TITLE
verify the SET when getting a log entry

### PR DESCRIPTION
**Pull request by asraa**
Signed-off-by: Asra Ali <asraa@google.com>

I noticed that in non-bundle, we are still using the integratedTime to check expiration without verifying rekor's signature of the signedEntryTimestamp

### CODE REVIEW

File: pkg/cosign/verify.go [hunk 1]
```diff
@@ -135,6 +135,16 @@ func VerifyTLogEntry(rekorClient *client.Rekor, uuid string) (*models.LogEntryAn
 	if err := v.VerifyInclusionProof(*e.Verification.InclusionProof.LogIndex, *e.Verification.InclusionProof.TreeSize, hashes, rootHash, leafHash); err != nil {
 		return nil, errors.Wrap(err, "verifying inclusion proof")
 	}
+
+	// Verify rekor's signature over the SET.
+	rekorPubKey, err := PemToECDSAKey([]byte(rekorPub))
```
Comment (asraa) [inline pkg/cosign/verify.go:140]: should i just get this from rekorClient instead of using the offline one?

Comment (priyawadhwa) [reply pkg/cosign/verify.go:140]: personally i think this is fine

**Review by priyawadhwa — APPROVED** [review]

File: pkg/cosign/verify.go (hunk 1 above)
Comment (asraa) [reply pkg/cosign/verify.go:140]: agreed! i actually might change it over so we don't have to use the prod instance in the e2e test (which is why it's failing)

**Comment by asraa** [comment]
Debugging e2e test. Sees to work fine with a prod rekor instance, but not with the test/local one. Working on it!

**Comment by asraa** [comment]
OK it fails because rekor doesn't add a signature on on the signed entry timestamp when you get an existing entry. i don't know why we don't maintain the signature?

**Comment by asraa** [comment]
Depends on the linked PR, will update when ready for review

**Comment by asraa** [comment]
OK! PR is merged, this is fixed now!

**Review by dlorenc — APPROVED** [review]

**Pull request merged**